src/parse/rules/python_rules.build_defs

```diff
@@ -404,12 +409,32 @@ def python_wheel(name, version, hashes=None, package_name=None, outs=None,
     if not url_base:
         raise ParseError('python.wheel_repo is not set in the config, must pass repo explicitly '
                          'to python_wheel')
-    # Look for an arch-specific wheel first; in some cases there can be both (e.g. protobuf
-    # has optional arch-specific bits) and we prefer the one with the cool stuff.
-    url1 = '%s/%s-%s-${OS}_${ARCH}.whl' % (url_base, package_name, version)
-    url2 = '%s/%s-%s.whl' % (url_base, package_name, version)
+    urls = []
+    if name_scheme:
+        urls.append(name_scheme.format(url_base=url_base,
+                                       package_name=package_name,
+                                       version=version))
+    elif CONFIG.PYTHON_WHEEL_NAME_SCHEME:
+        urls.append(CONFIG.PYTHON_WHEEL_NAME_SCHEME.format(url_base=url_base,
+                                                           package_name=package_name,
+                                                           version=version))
+    else:
+        # Populate urls using a reasonable set of possible wheel naming schemes.
+        # Look for an arch-specific wheel first; in some cases there can be both (e.g. protobuf
+        # has optional arch-specific bits) and we prefer the one with the cool stuff.
+        urls.append('{url_base}/{package_name}-{version}-${{OS}}-${{ARCH}}.whl'.format(url_base=url_base,
```

Comment:
Do you still need this version as well as the one beneath?

Reply:
Maybe not! I can definitely customize for our wheel naming schema using the functionality in this PR, I just thought it might make sense to leave this as one of the default schemas since it's the standard according to [wheel's](https://pypi.python.org/pypi/wheel) documentation.

It might have a speed cost for your use-case though, if it has to check that format first, not find it, then fallback to the thought-machine schema.

So... I'm okay with whatever you prefer, since it's firstly a thought-machine project :)

Reply:
Yeah, I was thinking of a speed cost but it's not very significant - by the design of the system we don't have to re-request those packages very often. Makes sense to leave as is.